packages/flutter/lib/src/material/drawer.dart

```diff
@@ -250,6 +250,13 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
     super.dispose();
   }
 
+  @override
+  void didUpdateWidget(DrawerController oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.scrimColor != oldWidget.scrimColor)
+      _color = ColorTween(begin: Colors.transparent, end: widget.scrimColor ?? Colors.black54,);
```

Comment:
Suggestion: Move black54 to a const since it is used twice, or perhaps extract this whole line to a small function since it is duplicated twice. Your call.

Reply:
Good idea, I moved the code to _buildScrimColorTween()